### PR TITLE
Add support for macOS #233

### PR DESCRIPTION
**Pull request by ThilinaTCH**
`flutter config --enable-macos-desktop
`
`flutter create --org com.cybear_jinni.smart_home .`

Added following lines to info.plist where the string values represent the reason you need to access the user's location
```
<key>NSLocationWhenInUseUsageDescription</key>
<string>This application requires location services to work</string>

<key>NSLocationAlwaysUsageDescription</key>
<string>This application requires location services to work</string>
```

macOS needs a specific entitlement in order to access the network. To do that edited `macos/Runner/DebugProfile.entitlements` by adding the following key-value pair.

```
<key>com.apple.security.network.client</key>
<true/>
```
Then did the same thing in `macos/Runner/Release.entitlements`.

# Screenshots

Screenshot 2021-10-05 at 8 10 58 PM

Screenshot 2021-10-05 at 8 11 10 PM

### CODE REVIEW

**Review by guyluz11 — COMMENTED** [review]
Please update the pull request with the following comments and answer the questions about some parts.

File: macos/Runner/Assets.xcassets/AppIcon.appiconset/Contents.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
+{
+  "images" : [
+    {
+      "size" : "16x16",
+      "idiom" : "mac",
+      "filename" : "app_icon_16.png",
```
[the remark this answers is not in the document]
Comment (ThilinaTCH) [reply macos/Runner/Assets.xcassets/AppIcon.appiconset/Contents.json:6]: Yeah sure, please create an issue and assign it to me

File: macos/Runner/Configs/AppInfo.xcconfig [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Application-level settings for the Runner target.
+//
+// This may be replaced with something auto-generated from metadata (e.g., pubspec.yaml) in the
+// future. If not, the values below would default to using the project name when this becomes a
+// 'flutter create' template.
+
+// The application's name. By default this is also the title of the Flutter window.
+PRODUCT_NAME = cbj_app
```
[the remark this answers is not in the document]
Comment (ThilinaTCH) [reply macos/Runner/Configs/AppInfo.xcconfig:8]: Changed and can see it in app title no
Screenshot 2021-10-06 at 5 12 42 AM
w

File: macos/Runner/Configs/AppInfo.xcconfig [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+// The application's bundle identifier
+PRODUCT_BUNDLE_IDENTIFIER = com.cybearjinni.smarthome.cbjApp
```
[the remark this answers is not in the document]
Comment (ThilinaTCH) [reply macos/Runner/Configs/AppInfo.xcconfig:11]: Done!

**Pull request merged**